moralization/tests/test_spacy_model.py

```diff
@@ -0,0 +1,125 @@
+from moralization.spacy_model import Spacy_Setup, Spacy_Training
+from tempfile import mkdtemp
+import pathlib
+import pytest
+from shutil import copy
+
+
+def test_Spacy_Setup(data_dir):
+    tmp_dir = mkdtemp()
+
+    # test datadir and specific file path
+    Spacy_Setup(data_dir)
+
+    # test data_dir, working_dir and specific file.
+    Spacy_Setup(data_dir, working_dir=tmp_dir)
+
+    # test finding file by name in data_dir
+    Spacy_Setup(
+        pathlib.Path(__file__).parents[2].resolve() / "data" / "Training",
+    )
+    Spacy_Setup(
+        pathlib.Path(__file__).parents[2].resolve() / "data" / "Training",
+        working_dir=tmp_dir,
+    )
+
+
+def test_Spacy_Setup_convert_data_to_spacy(data_dir):
+
+    # test datadir and specific file path
+    test_setup = Spacy_Setup(data_dir)
+    test_setup.convert_data_to_spacy_doc()
+    assert (
+        list(test_setup.doc_dict.keys()).sort()
```

Comment:
There is a bug here that I am fixing now:
1. Calling `.sort()` directly in the `assert` statement does not work, because `.sort()` sorts the list in-place and returns `None`. So
```
test = ["z", "c"]
print(test.sort())
```
will print `None` (and I believe the `assert` will always return `True` in the code as you are then comparing `None` to `None`, also as highlighted by Sonarcloud as one of the issues).
But:
```
test = ["z", "c"]
test.sort()
print(test)
```
will work correctly, `test` has been sorted in-place.
So in this case, you would need to assign temporary variables to those lists.

You can avoid that using `sorted()`.